KHLY2700/KHLY2700-IOC-01App/Db/keithley2700_channels.template

```diff
@@ -42,11 +46,87 @@ record(ao, "$(P)$(CHANNEL):TIME")
 {
 	field(DESC, "Latest device reading timestamp")
 	field(EGU, "")
+	field(PREC, "7")
+	info(INTEREST, "HIGH")
+	info(ARCHIVE, "VAL")
+	field(FLNK, "$(P)$(CHANNEL):TEMP:RAW")
+}
+
+record(ao, "$(P)$(CHANNEL):TIME:PREV")
+{
+	field(DESC, "Previous timestamp value")
+	field(EGU, "")
+	field(PREC, "5")
+}
+
+record(cvt, "$(P)$(CHANNEL):TEMP:RAW")
+{
+  field(DESC, "Calculate temperature from calib file")
+  field(SCAN, "Passive")
+  field(PREC, "5")
+  field(EGU, "K")
+  field(INPY, "$(P)$(CHANNEL):READ")
+  field(METH, "1D TABLE INVERTED")
+  field(DRVL, "$(DRVLO)")
+  field(DRVH, "$(DRVHI)")
+  field(SPEC, "$(CALIB_FILE)")
+  field(TDIR, "$(CALIB_DIR)")
+  field(BDIR, "$(CALIB_BASE_DIR)")
+  info(INTEREST, "LOW")
+  field(FLNK, "$(P)$(CHANNEL):TEMP:CHECK")
+}
+
+record(calcout, "$(P)$(CHANNEL):TEMP:CHECK")
+{
+	field(DESC, "Check temp is within range")
+	field(OOPT, "Every Time")
+	field(INPA, "$(P)$(CHANNEL):TEMP:RAW")
+	field(CALC, "((A>=0)&&(A<320))?A:0")
+	field(OUT, "$(P)$(CHANNEL):TEMP PP")
+	field(FLNK, "$(P)$(CHANNEL):DRIFT:CALC")
+}
+
+record(ao, "$(P)$(CHANNEL):TEMP")
+{
+	field(DESC, "Temp value post range check")
+	field(EGU, "K")
 	field(PREC, "5")
 	info(INTEREST, "HIGH")
 	info(ARCHIVE, "VAL")
 }
 
+record(ao, "$(P)$(CHANNEL):TEMP:PREV")
+{
+	field(DESC, "Previous temperature value")
+	field(EGU, "K")
+	field(PREC, "5")
+}
+
+record(aSub, "$(P)$(CHANNEL):DRIFT:CALC")
+{
+	field(DESC, "Calculate drift")
+	field(SNAM, "calculate_drift")
+	field(NOA, "1")
+	field(INPA, "$(P)$(CHANNEL):TEMP")
+	field(INPB, "$(P)$(CHANNEL):TEMP:PREV")
+	field(INPC, "$(P)$(CHANNEL):TIME")
+	field(INPD, "$(P)$(CHANNEL):TIME:PREV")
+	field(INPE, "$(P)$(CHANNEL):DRIFT")
+	field(OUTA, "$(P)$(CHANNEL):DRIFT PP")
+	field(OUTB, "$(P)$(CHANNEL):TEMP:PREV PP")
+	field(OUTC, "$(P)$(CHANNEL):TIME:PREV PP")
+}
+
+record(ao, "$(P)$(CHANNEL):DRIFT")
+{
+	field(DESC, "Drift value")
+	field(PREC, "6")
+	field(EGU, "")
+	info(INTEREST, "HIGH")
+	info(ARCHIVE, "VAL")
+  field(TPRO,  "1")
```

Comment:
This is for debugging and should not be in there.

Reply:
removed